primeqa/pipelines/components/reader/generative.py

```diff
@@ -24,3 +32,118 @@ def load(self, *args, **kwargs):
 
     def apply(self, input_texts: List[str], context: List[List[str]], *args, **kwargs):
         pass
+    
+@dataclass
+class GenerativeFiDReader(ReaderComponent):
+    
+    model: str = field(
+        default="PrimeQA/eli5-fid-bart-large-with-colbert-passages",
+        metadata={"name": "Model"},
+    )
+    max_seq_len: int = field(
+        default=256,
+        metadata={
+            "name": "Maximum sequence length",
+            "description": "Maximum length of question and context inputs to the model (in word pieces/bpes)",
+            "range": [32, 512, 8],
+        },
+    )
+    max_answer_length: int = field(
```

Comment:
What is the difference between max_answer_length and generation_max_length?

Reply:
They are redundant but the max_answer_length was used in the postprocessor while the trainer needs this generation_max_length which is a trainer argument from HF.

Reply:
Can you just set one using the other in the init function?